stablehlo/dialect/AssemblyFormat.h

```diff
@@ -218,7 +222,6 @@ ParseResult parseCustomCallTarget(AsmParser& parser, StringAttr& target);
 //     [1, 2]
 void printIntArray(AsmPrinter& printer, ArrayRef<int64_t> ints);
 
-FailureOr<SmallVector<int64_t>> parseIntArray(AsmParser& parser);
 ParseResult parseIntArray(AsmParser& parser, SmallVector<int64_t>& ints);
```

Comment:
`parseIntArray` is only used in `parseDims` within StablehloOps.cpp. Given that, I think we should delete `parseIntArray` from `AssemblyFormat.h`.

As far as what happens with the implementation of `parseIntArray`, I think that we have two options:
  1) Inline the implementation into `parseDims`.
  2) Delete `parseDims` and replace its usages with `parseDimSizes`.

I think that 1) is more consistent with this comment in #327: https://github.com/openxla/stablehlo/issues/327#issuecomment-1319353977, i.e. I'm supportive of deferring the cleanup of `parseDims` to future PRs. I think that we should delete `parseIntArray` right away though.

Reply:
There was no fallout from (2) in StableHLO so let's make that change now. If there is any fallout from MHLO (not sure there will be, looks like a lot of the uses of parseDim should only specify positive dim numbers), we can inline into a parseDims implementation there and clean up later.